fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.ts

```diff
@@ -1060,35 +1060,48 @@ export class FmcAircraftInterface {
       }
     }
 
-    // Retrieve AOA from ADRs
+    // Retrieve AOA, CAS and altitude from ADRs
     const aoa1 = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_1_ANGLE_OF_ATTACK');
     const aoa2 = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_2_ANGLE_OF_ATTACK');
     const aoa3 = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_3_ANGLE_OF_ATTACK');
     const cas1 = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_1_COMPUTED_AIRSPEED');
     const cas2 = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_2_COMPUTED_AIRSPEED');
     const cas3 = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_3_COMPUTED_AIRSPEED');
+    const alt1 = Arinc429Register.empty().setFromSimVar('L:A32NX_ADIRS_ADR_1_ALTITUDE');
+    const alt2 = Arinc429Register.empty().setFromSimVar('L:A32NX_ADIRS_ADR_2_ALTITUDE');
+    const alt3 = Arinc429Register.empty().setFromSimVar('L:A32NX_ADIRS_ADR_3_ALTITUDE');
     const adrOp = [aoa1.isNormalOperation(), aoa2.isNormalOperation(), aoa3.isNormalOperation()];
     const casOp = [cas1.isNormalOperation(), cas2.isNormalOperation(), cas3.isNormalOperation()];
+    const altOp = [alt1.isNormalOperation(), alt2.isNormalOperation(), alt3.isNormalOperation()];
 
     let aoa = 0;
     let cas = 0;
+    let alt = 0;
     if (adrOp.filter(Boolean).length === 3) {
       const aoaSorted = [aoa1.value, aoa2.value, aoa3.value].sort((a, b) => a - b);
       const casSorted = [cas1.value, cas2.value, cas3.value].sort((a, b) => a - b);
+      const altSorted = [alt1.value, alt2.value, alt3.value].sort((a, b) => a - b);
       aoa = aoaSorted[1];
       cas = casSorted[1];
+      alt = altSorted[1];
     } else if (adrOp.filter(Boolean).length === 2) {
       const aoas = [aoa1, aoa2, aoa3].filter((v, i) => adrOp[i]);
       aoa = (aoas[0].value + aoas[1].value) / 2;
 
       const cass = [cas1, cas2, cas3].filter((v, i) => casOp[i]);
       cas = (cass[0].value + cass[1].value) / 2;
+
+      const alts = [alt1, alt2, alt3].filter((v, i) => altOp[i]);
+      alt = (alts[0].value + alts[1].value) / 2;
     } else if (adrOp.filter(Boolean).length === 1) {
       const aoas = [aoa1, aoa2, aoa3].filter((v, i) => adrOp[i]);
       aoa = aoas[0].value;
 
       const cass = [cas1, cas2, cas3].filter((v, i) => casOp[i]);
       cas = cass[0].value;
+
+      const alts = [alt1, alt2, alt3].filter((v, i) => altOp[i]);
+      alt = alts[0].value;
```

Comment:
Why not just use the FMGC Navigation class for this? It's job is to source navigation data from the relevant systems, select the appropriate sources, and eventually also run the filters to combine sources where applicable.

Reply:
With the next commit, CAS alt alt are going to be taken from NavigationProvider. AoA is going to be removed, not used anymore in the FMS calculations